thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/api/MetricTimeSeries.java

```diff
@@ -265,9 +281,28 @@ public String toString() {
     return result;
   }
 
+  public Integer[] getHasValueSums() {
+    Integer[] result = new Integer[schema.getNumMetrics()];
+
+    for (int i = 0; i < schema.getNumMetrics(); i++) {
+      result[i] = 0;
+    }
+
+    for (Long time : hasValue.keySet()) {
+      boolean[] booleans = hasValue.get(time);
+      for (int i = 0; i < schema.getNumMetrics(); i++) {
+        if (booleans[i]) {
+          result[i] = result[i] + 1;
+        }
+      }
+    }
+
+    return result;
+  }
+
   @Override
   public int hashCode() {
-    return timeseries.keySet().hashCode() + 13 * schema.hashCode();
+    return metricsValue.keySet().hashCode() + 13 * schema.hashCode();
```

Comment:
Is 13 a real 'magic' number?

Reply:
Yeah, it is a magic number, which is not a good idea to use it. Replacing it with Java7's Objects.hash().